text/0000-release-process.md

```diff
@@ -0,0 +1,296 @@
+# Meta
+[meta]: #meta
+- Name: Release Process
+- Start Date: 2019-11-18
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+The release process for the various components maintained by the Cloud Native Buildpacks organization.
+
+# Motivation
+[motivation]: #motivation
+
+Through the previous releases of pack and lifecycle we've come to learn that a time-based release cadence makes the
+most sense. Unfortunately, due to the fact that there is no formal process, making the release happen has been met by
+some inconsistencies in timing and confusion in processes. These are generally in the form of conversations revolving 
+the following questions _"when are we going to release?"_, _"what do we want to put in this release?"_, 
+_"should we wait for feature Y to release?"_, etc.
+
+# Consumer Impact
+
+#### Improvements
+
+At the moment, releases tend to get pushed back to ensure certain feature sets are released. With more
+definitive guidance we could plan around it and isolate features when necessary through the process of
+[CCB](#change-control-board) while maintaining the goal of releasing on time.
+
+#### Concerns
+
+If the [Release Cadence](#timetable) is too long, consumers would need to wait longer for features they request
+if they are not included in the currently scheduled release.
+
+# What it is
+[what-it-is]: #what-it-is
+
+The proposed release process aims to be flexible while providing structure. It proposes a 3-phase release cycle for
+the purpose of:
+ 
+1. Enabling [scheduled releases](#timetable)
+2. Allowing for the [continuation of development](#concurrency)
+3. [Isolating releases](#release-branches) and their features
+4. Ensure [user acceptance](#user-acceptance-testing) takes place
+5. [Migration](#migration-guide) is considered
+
+### Phases
+Each release would consists of the following 3 phases:
+
+- development - standard development process _(issue -> PR -> merge)_
+- [feature complete](#feature-complete) - branch of code into a [release branch](#release-branches) for ongoing [UAT](#user-acceptance-testing)
+- [release execution](#release-execution) - the step of finalizing and publishing a release
+
+```text
+ release X
+
+ +---------------+      +------------------+      +---------+
+ |  development  | ---> | feature complete | ---> | release |
+ +---------------+      +------------------+      +---------+
+
+```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+### Timetable
+
+_The following schedules are suggestive and subject to change based on practicality._
+
+* Release Cadence: **Monthly** on the **1st Tuesday** of the month 
```

Comment:
Are all releases happening on the 1st Tues? Does this only apply to `pack`, to `lifecycle`? What about if we want to release one of them earlier b/c of a breaking change, do we need to wait a month then?

Reply:
I also wonder if a monthly release cadence is too much churn?

Reply:
Updated the summary to present this as a _template_ that only pack and lifecycle would use initially. The specifics on dates are agreed upon outside this document.